cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1722,6 +1700,119 @@ pub(crate) fn emit(
             }
         }
 
+        Inst::GrowFrame { amount, tmp } => {
+            debug_assert!(*amount > 0);
+            debug_assert_eq!(*amount % 8, 0);
+
+            let tmp = allocs.next(tmp.to_reg().to_reg());
+            let src = Gpr::new(tmp).unwrap();
+            let dst = WritableGpr::from_reg(src);
+
+            // As we're increasing the number of stack arguments, we need to move the frame down in
+            // memory, by decrementing SP by `amount` and looping from lower addresses to higher
+            // ones, copying down.
+
+            // TODO: this needs to be accounted for by the stack check, before `GrowFrame` is
+            // emitted, so that the increment here is expected.
+            //
+            // Decrement SP and FP by `amount`
+            Inst::alu_rmi_r(
+                OperandSize::Size64,
+                AluRmiROpcode::Sub,
+                RegMemImm::imm(*amount),
+                Writable::from_reg(regs::rsp()),
+            )
+            .emit(&[], sink, info, state);
+            Inst::alu_rmi_r(
+                OperandSize::Size64,
+                AluRmiROpcode::Sub,
+                RegMemImm::imm(*amount),
+                Writable::from_reg(regs::rbp()),
+            )
+            .emit(&[], sink, info, state);
+
+            // The total size that we're going to copy, including the return address and frame
+            // pointer that are pushed on the stack alreadcy.
+            let size = i32::try_from(state.nominal_sp_to_fp()).unwrap() + 16;
```

Comment:
After we wrote this we discovered the `fp_to_arg_offset` method which I now realize we could also use here instead of hard-coding 16. That might also be handy for generalizing this in shared code when we extend this work to other targets.